client/web/wasm/src/viewport.rs

```diff
@@ -16,22 +16,35 @@ pub fn select_tool(tool: String) -> Result<(), JsValue> {
 	})
 }
 
-static mut POS: (u32, u32) = (0, 0);
-
 /// Mouse movement with the bounds of the canvas
 #[wasm_bindgen]
 pub fn on_mouse_move(x: u32, y: u32) {
-	// SAFETY: This is safe because the code can only ever run in a single thread
-	unsafe {
-		POS = (x, y);
-	}
+	EDITOR_STATE.with(|editor| {
+		let mut editor = editor.borrow_mut();
+		if editor.tools.mouse_is_clicked {
+			editor.tools.trace.append_point(x, y)
+		}
+	})
 }
 
 /// Mouse click within the bounds of the canvas
 #[wasm_bindgen]
 pub fn on_mouse_click(x: u32, y: u32) -> Result<(), JsValue> {
 	let ev = events::Event::Click(events::MouseState::from_pos(x, y));
-	EDITOR_STATE.with(|editor| editor.borrow_mut().handle_event(ev)).map_err(|err| Error::new(&err.to_string()).into())
+	EDITOR_STATE
+		.with(|editor| {
+			let mut editor = editor.borrow_mut();
+			editor.tools.mouse_is_clicked = true;
+			editor.tools.trace.clear();
+			editor.handle_event(ev)
```

Comment:
Maybe we should rename the event since it sounds kind of weird that each point in a trace will send a `Click` event